### ISSUE TITLE
sock_read timeout's behavior differ from documentation

### ISSUE THREAD

**Issue by GoPavel** [aio-libs/aiohttp#3808]
## Problem

It's stated in the [doc](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientTimeout.sock_connect) that `sock_timeout` for `aiohttp.web.StreamResponse` limits delay between chunks. So I think the following test should be passed:
```python3
async def test_read_timeout_between_chunks(aiohttp_client) -> None:
    async def handler(request):
        resp = aiohttp.web.StreamResponse()
        await resp.prepare(request)
        for _ in range(4):
            await asyncio.sleep(0.1)
            await resp.write(b'data\n')
        return resp

    app = web.Application()
    app.add_routes([web.get('/', handler)])

    timeout = aiohttp.ClientTimeout(sock_read=0.2)
    client = await aiohttp_client(app, timeout=timeout)

    res = b''
    async with await client.get('/') as resp:
        res += await resp.read()

    assert res == b'data\n' * 4

```
However now it's failing with `ServerTimeoutError`. `sock_read` is the de facto timeout for all readings from `aiohttp.web.StreamResponse`.

## Your environment

Fedora 30, x86_64 Linux 5.0.17-300.fc30.x86_64
python: 3.7.4
aiohttp: v3.6.0 (b33540c)

**Comment by asvetlov** [comment]
@mjpieters please take a look.
Is it related to the problem that we discussed on gitter recently?

**Comment by mjpieters** [comment]
This does look exactly like the test I wanted to write for the issue I see.

@GoPavel: you should be able to resolve this by adding `self._reschedule_timeout()` as the first line of [`aiohttp/client/client_proto.py:ResponseHandler.data_received()`](https://github.com/aio-libs/aiohttp/blob/b33540cafa432503baa517d0fd76abf67b647598/aiohttp/client_proto.py#L178)

**Comment by mjpieters** [comment]
This was included in the new [v3.6.2 release](https://github.com/aio-libs/aiohttp/releases/tag/v3.6.2).
